plugin/evm/vm.go

```diff
@@ -314,14 +308,26 @@ type VM struct {
 	rpcHandlers []interface{ Stop() }
 }
 
+func newExtensibleEVM(isPlugin bool) *sharedEvm {
```

Comment:
could we have a function like newVM that takes the `codec.Manager` and drop the SetNetworkCodec function below? 
It seems this type is not fully initialized even if it is constructed with a "new" function.